server/models/ParticipantsModel/index.js

```diff
@@ -29,33 +29,20 @@ const ParticipantsModel = {
         .collection(collections.assessmentDayData)
         .find(query, { projection: ALL_SUBJECTS_MONGO_PROJECTION })
         .stream()
-    } else {
-      const mongoFacet = filtersService.barChartMongoQueries()
-      const isAggregateQuery = Array.isArray(mongoFacet)
-      const query = isAggregateQuery
-        ? mongoFacet
-        : Object.keys(mongoFacet).reduce((_, next) => {
-            return mongoFacet[next][0].$match
-          }, {})
-
-      const participants = isAggregateQuery
-        ? await AssessmentDayDataModel.index(db, query)
-        : await AssessmentDayDataModel.all(db, query)
-
-      const chartQuery = {
-        assessment: chart.assessment,
-        participant: {
-          $in: isAggregateQuery
-            ? participants[0].participants
-            : participants.map(({ participant }) => participant),
-        },
-      }
-
-      return await db
-        .collection(collections.assessmentDayData)
-        .find(chartQuery, { projection: ALL_SUBJECTS_MONGO_PROJECTION })
-        .stream()
     }
+
+    const includedParticipants = await db.collection(collections.assessmentDayData)
+      .distinct('participant', {
+        $or: filtersService.filterQueries
+      })
+    return await db
+      .collection(collections.assessmentDayData)
+      .aggregate([
+        { $match: { study: { $in: filters.sites, $nin: STUDIES_TO_OMIT }} },
+        { $match: { participant: { $in: includedParticipants }}},
+        { $match: { assessment: chart.assessment }}
+      ])
```

Comment:
It's non-blocking suggestion.
```suggestion
    return await db
      .collection(collections.assessmentDayData)
      .aggregate([
        { $match: { assessment: chart.assessment ,study: { $in: filters.sites, $nin: STUDIES_TO_OMIT }} },
        { $match: { participant: { $in: includedParticipants }}},
       ])
```